connector/dao/src/main/java/it/pagopa/selfcare/mscore/connector/dao/UserConnectorImpl.java

```diff
@@ -70,6 +72,19 @@ public List<OnboardedUser> findAll() {
         return repository.findAll().stream().map(userMapper::toOnboardedUser).collect(Collectors.toList());
     }
 
+    @Override
+    public List<OnboardedUser> findAll(Integer page, Integer size, String productId) {
+        Pageable pageable = PageRequest.of(page, size);
+        Query queryMatch = Query.query(Criteria.where(UserEntity.Fields.bindings.name())
+                .elemMatch(Criteria.where(UserBinding.Fields.products.name())
+                        .elemMatch(Criteria.where(OnboardedProductEntity.Fields.productId.name()).is(productId)
+                                .and(OnboardedProductEntity.Fields.status.name()).in(List.of(RelationshipState.ACTIVE.name(), RelationshipState.DELETED.name(), RelationshipState.SUSPENDED.name())))));
```

Comment:
can static List have better performance?

Reply:
Thought about using unwind operation on mongo, but the performance level it might be more slow, better retrieve the whole list and filter it with Java